### ISSUE TITLE
From pyiron.atomistics import Project does not work.

### ISSUE THREAD

**Issue by niklassiemer** [pyiron/pyiron#1302]
I am not sure how we want to handle this, however, if I am not mistaken we want to support the nomenclature `from pyiron.atomistics import Project`. Right now, we bind the `atomistics` module of `pyiron_atomistics` to `pyiron.atomistics` to be backwards compatible. Any thoughts about how to resolve this naming conflict? 

https://github.com/pyiron/pyiron/blob/40f43e069bd963bea2c48602220b5b2df0776186/pyiron/__init__.py#L16-L19

**Comment by niklassiemer** [comment]
Could we copy the API of the global `pyiron_atomistics.__init__` to the `pyiron_atomistics.atomistics` such that we allow for the new type of initialization of the `Project` for a temporary fix until we deprecate `from pyiron import Projecy`?

**Comment by jan-janssen** [comment]
Is not the underlying issue: 
```
pyiron_atomistics.atomistics = pyiron.atomistics
```
Then it might be better to fix: 
```
sys.modules["pyiron.atomistics"] = pyiron_atomistics
```
But this would again break the backwards compatibility.

**Comment by niklassiemer** [comment]
Yes, that's exactly our problem... I just thought about a way to not instantly break backwards compatibility and to warn the users before we actually break it. However, the new syntax should also be working right now.
Therefore, copying the API was the only solution I could think of to support
```
from pyiron import Project   #  give deprecation warning
from pyiron.atomistics import Project # The future behavior
from pyiron.atomistics.something # stay backwards compatible
```
However, we would only have the API of the new `pyiron.atomistics = pyiron_atomistics` reachable... 

Maybe we should from now on and for one or two releases give a deprecate-soon warning and how to do it in the future and then have a hard break with a deprecation error?
